### PR TITLE
Warn team about high failure rates

### PR DESCRIPTION
**Pull request by CrystalPea**
Log an exception and send a zendesk ticket when, in the last 24 hours:
- there were services with failure rate over 25% for their sms with at least 100 messages sent
- there were services that sent at least 100 sms messages to tv numbers

Story ticket: https://www.pivotaltracker.com/n/projects/1443052

### CODE REVIEW

File: app/celery/scheduled_tasks.py [hunk 1]
```diff
@@ -253,3 +256,42 @@ def check_for_missing_rows_in_completed_jobs():
...
+        for service in services_sending_to_tv_numbers:
+            message += "service id: {}, count of sms to tv numbers: {},\n".format(
+                service.service_id, service.notification_count
+            )
+
+    if services_with_failures or services_sending_to_tv_numbers:
+        current_app.logger.exception(message)
+
+        if current_app.config['NOTIFY_ENVIRONMENT'] in ['live', 'production', 'test']:
+            message += "\nThings to do: contact service? revoke their key?"
```
Comment (CrystalPea) [inline app/celery/scheduled_tasks.py]: Explain what TV numbers are:
These are numbers reserved for the use in TV and drama. The range of TV numbers are 07700 900000 to 900999. When an SMS notification is sent to such a number, we get a 'permanent-failure' status back and we are billed for this notification.

Comment (idavidmcdonald) [reply app/celery/scheduled_tasks.py]: Thanks. That first sentence would be useful to put in the TODO section or in the runbook so a developer receiving the ticket knows how to deal with it

Comment (CrystalPea) [reply app/celery/scheduled_tasks.py]: yeah, I wrote it to myself here only to remember about including it :). As it only transpired in a spoken conversation.

**Comment by CrystalPea** [comment]
@servingUpAces I refactored this code as per our conversations now, what do you think? 

@idavidmcdonald I created a section for this ticket in the Manual and linked it from the ticket.

**Comment by idavidmcdonald** [comment]
@CrystalPea Excellent runbook entry :)

File: tests/app/dao/test_services_dao.py [hunk 2]
```diff
@@ -1099,3 +1071,79 @@ def create_email_sms_letter_template():
     template_two = create_template(service=service, template_name='2', template_type='sms')
     template_three = create_template(service=service, template_name='3', template_type='letter')
     return template_one, template_three, template_two
+
+
+@freeze_time("2019-12-02 12:00:00.000000")
```
Comment (servingUpAces) [inline tests/app/dao/test_services_dao.py]: Do we need to freeze_time for this test?

File: tests/app/dao/test_services_dao.py [hunk 3]
```diff
@@ -1099,3 +1071,79 @@ def create_email_sms_letter_template():
...
+def test_dao_find_services_sending_to_tv_numbers(notify_db_session, fake_uuid):
+    service_1 = create_service(service_name="Service 1", service_id=fake_uuid)
```
Comment (servingUpAces) [inline tests/app/dao/test_services_dao.py]: Do we need to use fake_uuid?

File: tests/app/dao/test_services_dao.py [hunk 4]
```diff
@@ -1099,3 +1071,79 @@ def create_email_sms_letter_template():
...
+
+    service_6 = create_service(service_name="Service 6")  # notifications too old are excluded
+    with freeze_time("2019-11-30 15:00:00.000000"):
+        template_6 = create_template(service_6)
+        for x in range(0, 5):
+            create_notification(template_6, normalised_to=tv_number, status="permanent-failure")
+
+    service_2 = create_service(service_name="Service 2")  # below threshold is excluded
+    template_2 = create_template(service_2)
+    create_notification(template_2, normalised_to=tv_number, status="permanent-failure")
```
Comment (servingUpAces) [inline tests/app/dao/test_services_dao.py]: Should we create some `test` notifications?

File: tests/app/dao/test_services_dao.py (hunk 2 above)
Comment (CrystalPea) [reply tests/app/dao/test_services_dao.py]: I think we do, to show that most notifications are in good date range, while notifications for service_6 are just a bit too old and do not make it to results.

File: tests/app/dao/test_services_dao.py (hunk 3 above)
Comment (CrystalPea) [reply tests/app/dao/test_services_dao.py]: We need an uuid, as we assert on it below, to check the result we get is for the right service.

File: tests/app/dao/test_services_dao.py (hunk 4 above)
Comment (CrystalPea) [reply tests/app/dao/test_services_dao.py]: Good idea, I will add those.

File: tests/app/dao/test_services_dao.py (hunk 3 above)
Comment (servingUpAces) [reply tests/app/dao/test_services_dao.py]: You can just use the id from service_1.id

**Pull request merged**